pkg/crc/preflight/preflight_daemon_task_check_windows.go

```diff
@@ -112,8 +111,8 @@ func removeDaemonTask() error {
 }
 
 func checkIfDaemonTaskRunning() error {
-	if isDaemonRunningWithReleasedVersion() {
-		return nil
+	if err := olderDaemonVersionRunning(); err != nil {
+		return err
```

Comment:
This feels redundant with https://github.com/code-ready/crc/pull/3133 though. Maybe https://github.com/code-ready/crc/pull/3133 should remove that daemon version check, and then this PR becomes simpler?

Reply:
@cfergeau In windows we don't have autostart of task so that's why we have this redundant code, I can re-evaluate it after the release.

Reply:
Ah right still no autostart :( Imo this has to be cleaned up as soon as possible (but I agree not for this release), as we are handling too many things (daemon started through a task, daemon started manually, daemon running with out of date task file, out of date daemon running, ...)